### PR TITLE
refactor: Port card operation to view card version and enable setting

### PR DESCRIPTION
**Pull request by kartikCypherock**
This pull request enables viewing X1 card version. A new field `card_version` is added in `nfc_connection_data` structure, which can hold a pointer to buffer where the card version will be copied on card tap.

Fixes https://app.clickup.com/t/9002019994/PRF-4845

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4845 Card version](https://app.clickup.com/t/9002019994/PRF-4845)

File: src/card_operations/read_card_version.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                       const char *heading,
+                       const char *msg) {
+  bool result = false;
+  card_operation_data_t card_data = {0};
+  card_data.nfc_data.retries = 5;
+  card_data.nfc_data.card_version = card_version;
+
+  instruction_scr_init(msg, heading);
+
+  while (1) {
```
Comment (ujjwal-cyph) [inline src/card_operations/read_card_version.c:109]: can we remove the `while...` as the retriable situations are already handled by `card_initialize_applet`

Comment (kartikCypherock) [reply src/card_operations/read_card_version.c:110]: Hi Ujjwal, this while loop is required to handle only `CARD_OPERATION_CARD_REMOVED` error type, which would require a retap from the user..

Comment (ujjwal-cyph) [reply src/card_operations/read_card_version.c:110]: okay. makes sense

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**